### DISCUSSION TITLE
IPSEC connection is not working after the reboot

### DISCUSSION THREAD

**Discussion by dharmagondu** [strongswan/strongswan#1646] [Q&A]
The IPSEC connection is not working after node is rebooted. In order to restore it we need to restart racoon and racoon-setkey services on the node. Below are the configurations on the node:

ipsecls -l

IP Security Policies

Policy name          : IMSPolicy2                              
Local host address   : 198.18.25.166        --Cluster ipaddress                   
Remote host address  : 10.7.132.227                            
Mode                 : transport                               
IPSec protocol       : ESP                                     
Encryption algorithm : 3des                                    
Hash algorithm       : md5                                     
IKE phase 2 lifetime : 3600 sec                                
DH group             : 14                                      
PFS group            : 14                                      
Remote gateway       : na                                      

Policy name          : IMSPolicy                               
Local host address   : 198.18.25.166                           
Remote host address  : 10.29.66.115                            
Mode                 : transport                               
IPSec protocol       : ESP                                     
Encryption algorithm : 3des                                    
Hash algorithm       : md5                                     
IKE phase 2 lifetime : 3600 sec                                
DH group             : 14                                      
PFS group            : 14                                      
Remote gateway       : na                                      

# ls -l
total 28
- 1 root root 1214 Sep 19  2022 10.29.66.115
-rw-r--r-- 1 root root 1214 Sep 19  2022 10.29.66.115.racoon.conf
-rw-r--r-- 1 root root 1214 Mar 24  2021 10.7.132.227.racoon.conf
-rw-r--r-- 1 root root  544 Jan 23  2020 ipsec_backup.tar.gz
-r-------- 1 root root   60 Sep 19  2022 psk.txt
-r--r--r-- 1 root root  364 Sep 19  2022 racoon.conf
-rw-r--r-- 1 root root  961 Sep 19  2022 racoon_backup.tar.gz
-r--r--r-- 1 root root 1123 Sep 19  2022 setkey.conf

# cat racoon.conf
path pre_shared_key "/etc/racoon/psk.txt";

include "/etc/racoon/10.7.132.227.racoon.conf";
include "/etc/racoon/10.29.66.115.racoon.conf";
sainfo anonymous {
        pfs_group 2;
        encryption_algorithm 3des, aes, des, blowfish 448, twofish, rijndael;
        authentication_algorithm hmac_sha1, hmac_md5, hmac_sha256, hmac_sha384, hmac_sha512;
        compression_algorithm deflate;
}
# cat 10.29.66.115.racoon.conf
remote 10.29.66.115 {
    exchange_mode main;
    proposal {
            encryption_algorithm 3des;
            hash_algorithm md5;
            authentication_method pre_shared_key;
            dh_group 14;
    }
}

sainfo address 198.18.25.166/32 any address 10.29.66.115/32 any {
        pfs_group 14;
        lifetime time 3600 sec;
        encryption_algorithm 3des, aes, des, blowfish 448, twofish, rijndael;
        authentication_algorithm hmac_sha1, hmac_md5, hmac_sha256, hmac_sha384, hmac_sha512;
        compression_algorithm deflate;
}

sainfo address 198.18.25.164/32 any address 10.29.66.115/32 any {
        pfs_group 14;
        lifetime time 3600 sec;
        encryption_algorithm 3des, aes, des, blowfish 448, twofish, rijndael;
        authentication_algorithm hmac_sha1, hmac_md5, hmac_sha256, hmac_sha384, hmac_sha512;
        compression_algorithm deflate;
}

sainfo address 198.18.25.165/32 any address 10.29.66.115/32 any {
        pfs_group 14;
        lifetime time 3600 sec;
        encryption_algorithm 3des, aes, des, blowfish 448, twofish, rijndael;
        authentication_algorithm hmac_sha1, hmac_md5, hmac_sha256, hmac_sha384, hmac_sha512;
        compression_algorithm deflate;
}

When the connection is failing we can see cluster ipaddress in the racoon service status:
# systemctl status racoon  -l
;32mG— racoon.service - Racoon IKEv1 key management daemon for IPSEC
v
   Loaded: loaded (/usr/lib/systemd/system/racoon.service; disabled; vendor preset: disabled)
   Active: ;32mactive (running) since Tue 2023-04-04 15:15:20 CEST; 16s ago
 Main PID: 11847 (racoon)
    Tasks: 1 (limit: 512)
   CGroup: /system.slice/racoon.service
           G””G”€11847 /usr/sbin/racoon -F -v -f /etc/racoon/racoon.conf

Apr 04 15:15:27 SC-2-2 racoon[11847]: alg_oakley_hmacdef_one(hmac_md5 size=325): 0.000004
Apr 04 15:15:27 SC-2-2 racoon[11847]: ;39malg_oakley_hmacdef_one(hmac_md5 size=325): 0.000004
Apr 04 15:15:27 SC-2-2 racoon[11847]: ;39malg_oakley_hmacdef_one(hmac_md5 size=309): 0.000003
Apr 04 15:15:27 SC-2-2 racoon[11847]: alg_oakley_hmacdef_one(hmac_md5 size=325): 0.000004
Apr 04 15:15:27 SC-2-2 racoon[11847]: ;39malg_oakley_hmacdef_one(hmac_md5 size=325): 0.000004
Apr 04 15:15:27 SC-2-2 racoon[11847]: ;39malg_oakley_hmacdef_one(hmac_md5 size=325): 0.000004
Apr 04 15:15:27 SC-2-2 racoon[11847]: phase2(quick I msg2): 0.003828
Apr 04 15:15:27 SC-2-2 racoon[11847]: 2023-04-04 15:15:27: INFO: IPsec-SA established: ESP/Transport 198.18.25.166[500]->10.29.66.115[500] spi=46464402(0x2c4fd92)
Apr 04 15:15:27 SC-2-2 racoon[11847]: phase2(quick): 0.010682
Apr 04 15:15:27 SC-2-2 racoon[11847]: 2023-04-04 15:15:27: INFO: IPsec-SA established: ESP/Transport **198.18.25.166**[500]->10.29.66.115[500] spi=3346068259(0xc770f323)

After restarting the racoon and racoon-setkey services the IPSEC connection is restored and if we observe racoon status then we can see node ipaddress and not the cluster ipaddress:

# systemctl status racoon  -l
;32mG— racoon.service - Racoon IKEv1 key management daemon for IPSEC
vendor
   Loaded: loaded (/usr/lib/systemd/system/racoon.service; disabled; vendor preset: disabled)
   Active: ;32mactive (running) since Tue 2023-04-04 15:22:01 CEST; 46s ago
 Main PID: 18177 (racoon)
    Tasks: 1 (limit: 512)
   CGroup: /system.slice/racoon.service
           G””G”€18177 /usr/sbin/racoon -F -v -f /etc/racoon/racoon.conf

pr 04 15:22:17 SC-2-2 racoon[18177]: 
Apr 04 15:22:17 SC-2-2 racoon[18177]: alg_oakley_hmacdef_one(hmac_md5 size=325): 0.000004
Apr 04 15:22:17 SC-2-2 racoon[18177]: alg_oakley_hmacdef_one(hmac_md5 size=325): 0.000005
Apr 04 15:22:17 SC-2-2 racoon[18177]: ;39malg_oakley_hmacdef_one(hmac_md5 size=309): 0.000005
Apr 04 15:22:17 SC-2-2 racoon[18177]: alg_oakley_hmacdef_one(hmac_md5 size=325): 0.000006
Apr 04 15:22:17 SC-2-2 racoon[18177]: ;39malg_oakley_hmacdef_one(hmac_md5 size=325): 0.000021
Apr 04 15:22:17 SC-2-2 racoon[18177]: ;39malg_oakley_hmacdef_one(hmac_md5 size=325): 0.000005
Apr 04 15:22:17 SC-2-2 racoon[18177]: ;39mphase2(quick I msg2): 0.004411
Apr 04 15:22:17 SC-2-2 racoon[18177]: 2023-04-04 15:22:17: INFO: IPsec-SA established: ESP/Transport 198.18.25.165[500]->10.29.66.115[500] spi=227311474(0xd8c7f72)
Apr 04 15:22:17 SC-2-2 racoon[18177]: phase2(quick): 0.011912
Apr 04 15:22:17 SC-2-2 racoon[18177]: 2023-04-04 15:22:17: INFO: IPsec-SA established: ESP/Transport **198.18.25.165**[500]->10.29.66.115[500] spi=3253246529(0xc1e89a41)

Where, 198.18.25.164 is nodeA ip, 198.18.25.165 is nodeB ip and 198.18.25.166 is cluster ip.

Could you please help in identifying the rootcause.

**Comment by tobiasbrunner** [comment]
Why are you reporting this here? strongSwan has nothing to do with racoon/setkey.

**Reply by dharmagondu** [reply]
Could you please let me know the forum where it can be addressed.

**Reply by tobiasbrunner** [reply]
[racoon/ipsec-tools hasn't been maintained for years](https://ipsec-tools.sourceforge.net/). So you better use a different IKE implementation, e.g. strongSwan.
